docs/blog/posts/introduction-to-bpfd.md

```diff
@@ -0,0 +1,611 @@
+---
+date: 2023-09-07
+authors:
+  - anfredette
+  - shaneutt
+---
+
+# bpfd: A Novel Way to Manage eBPF
+
+In today's cloud ecosystem, there's a demand for low-level system access to
+enable high-performance observability, security, and networking functionality
+for applications. Historically these features have been implemented in
+[user space], however, the ability to program such functionality into the kernel
+itself can provide many benefits including (but not limited to) performance.
+Regardless, many Linux users still opt away from in-tree or [kernel
+module][kmod] development due to the slow rate of iteration and ensuing large
+management burden.  [eBPF] has emerged as a technology in the [Linux
+Kernel][Linux] looking to change all that.
+
+<!-- more -->
+
+eBPF is a simple and efficient way to dynamically load programs into the kernel
+at runtime, with safety and performance provided by the kernel itself using a
+Just-In-Time (JIT) compiler and verification process. There are a wide variety
+of program types one can create with eBPF, which include everything from
+networking applications to security systems.
+
+However, eBPF is still a fairly nascent technology and it's not all kittens and
+rainbows. The process of developing, testing, deploying, and maintaining eBPF
+programs is not a road well traveled yet, and the story gets even more
+complicated when you want to deploy your programs in a multi-node system, such
+as a [Kubernetes] cluster. It was these kinds of problems that motivated the
+creation of [bpfd], a system daemon for loading and managing eBPF programs in
+both traditional systems and Kubernetes clusters. In this blog post, we'll
+discuss the problems bpfd can help solve, and how to deploy and use it.
+
+[user space]:https://en.wikipedia.org/wiki/User_space_and_kernel_space
+[kmod]:https://wiki.archlinux.org/title/Kernel_module
+[eBPF]:https://ebpf.io
```

Comment:
Should eBPF just link to https://en.wikipedia.org/wiki/EBPF?

Reply:
I think that's fine. https://www.kernel.org/doc/html/latest/bpf/index.html would be fine too, but maybe for the target audience wikipedia is better.

Reply:
I changed it to Wikipedia.